### ISSUE TITLE
Grafana Upgrade from 8.3.4 to 9.0.6 breaks Mixed datasources using default uid

### ISSUE THREAD

**Issue by ChristopherzwTan** [grafana/grafana#54113]
**What happened**:
Upgraded Grafana from version 8.3.4 to 9.0.6. Panels that were using a "Mixed" datasource plugin with multiple default uid datasources did not upgrade properly and the panel becomes broken.

Grafana 8.3.4 Panel JSON snippet:
```
"targets": [
    {
      "datasource": "default",
      "hide": false,
      "refCount": 0,
      "refId": "A",
      "target": "lsg.*.online_users.*.0",
      "textEditor": true
    }
```

Grafana 9.0.6 Panel JSON snippet:
```
"targets": [
    {
      "datasource": {
        "type": "datasource",
        "uid": "-- Mixed --"
      },
      "hide": false,
      "refCount": 0,
      "refId": "A",
      "target": "lsg.*.online_users.*.0",
      "textEditor": true
    }
```

Screenshots:
Grafana 8 (before):
![grafana8-screenshot](https://user-images.githubusercontent.com/5447843/186232628-1857a357-87fe-4cbc-81a0-633e67e4e47f.png)

Grafana 9 (after):
![grafana9-screenshot](https://user-images.githubusercontent.com/5447843/186232641-c66f1681-77f7-4e2a-a506-a0904c7915ef.png)

**What you expected to happen**:
Upgrade should not have broken the panel. The JSON ideally should remain the same as `uid: "default"` and not `uid: "-- Mixed --"`

**How to reproduce it (as minimally and precisely as possible)**:
1. Set up Grafana 8.3.4 instance
2. Make a basic dashboard with a panel
3. Panel should use a mixed datasource, with a number of default datasource
4. Save panel and dashboard
5. Upgrade to Grafana 9.0.6

**Anything else we need to know?**:

**Environment**:
- Grafana version: 8.3.4 upgrade to 9.0.6
- Data source type & version: 
- OS Grafana is installed on: Grafana container
- User OS & Browser: macOS Monteray 12.5.1 and Google Chrome Version 104.0.5112.101 (Official Build) (x86_64)
- Grafana plugins: 
- Others:

**Comment by dprokop** [comment]
I can confirm this is a bug. It's still present on main. 

It's easy to reproduce. 
1. Run graphite docker block (from devenv dir run  `./create_docker_compose.sh graphite`)
2. Setup Graphite as default data source.
3. Import the attached dashboard[1].

<details>
<summary>
dashboard.json
</summary>

```json
{
    "annotations": {
      "list": [
        {
          "builtIn": 1,
          "datasource": "-- Grafana --",
          "enable": true,
          "hide": true,
          "iconColor": "rgba(0, 211, 255, 1)",
          "name": "Annotations & Alerts",
          "target": {
            "limit": 100,
            "matchAny": false,
            "tags": [],
            "type": "dashboard"
          },
          "type": "dashboard"
        }
      ]
    },
    "editable": true,
    "fiscalYearStartMonth": 0,
    "graphTooltip": 0,
    "id": 1,
    "links": [],
    "liveNow": false,
    "panels": [
      {
        "datasource": {
          "type": "datasource",
          "uid": "-- Mixed --"
        },
        "fieldConfig": {
          "defaults": {
            "color": {
              "mode": "palette-classic"
            },
            "custom": {
              "axisLabel": "",
              "axisPlacement": "auto",
              "barAlignment": 0,
              "drawStyle": "line",
              "fillOpacity": 0,
              "gradientMode": "none",
              "hideFrom": {
                "legend": false,
                "tooltip": false,
                "viz": false
              },
              "lineInterpolation": "linear",
              "lineWidth": 1,
              "pointSize": 5,
              "scaleDistribution": {
                "type": "linear"
              },
              "showPoints": "auto",
              "spanNulls": false,
              "stacking": {
                "group": "A",
                "mode": "none"
              },
              "thresholdsStyle": {
                "mode": "off"
              }
            },
            "mappings": [],
            "thresholds": {
              "mode": "absolute",
              "steps": [
                {
                  "color": "green",
                  "value": null
                },
                {
                  "color": "red",
                  "value": 80
                }
              ]
            }
          },
          "overrides": []
        },
        "gridPos": {
          "h": 9,
          "w": 12,
          "x": 0,
          "y": 0
        },
        "id": 2,
        "options": {
          "legend": {
            "calcs": [],
            "displayMode": "list",
            "placement": "bottom"
          },
          "tooltip": {
            "mode": "single"
          }
        },
        "targets": [
          {
            "datasource": "default",
            "refCount": 0,
            "refId": "A",
            "target": "carbon.agents.08202214a98a-a.cache.size"
          },
          {
            "datasource": "default",
            "hide": false,
            "refCount": 0,
            "refId": "B",
            "target": "carbon.agents.08202214a98a-a.cache.queues"
          },
          {
            "datasource": "default",
            "hide": false,
            "refCount": 0,
            "refId": "C",
            "target": "carbon.agents.08202214a98a-a.cache.queries"
          },
          {
            "datasource": "default",
            "hide": false,
            "refCount": 0,
            "refId": "D",
            "target": "carbon.agents.08202214a98a-a.cache.overflow"
          },
          {
            "datasource": "default",
            "hide": false,
            "refCount": 0,
            "refId": "E",
            "target": "carbon.agents.08202214a98a-a.cache.bulk_queries"
          }
        ],
        "title": "Panel Title",
        "type": "timeseries"
      }
    ],
    "refresh": "",
    "schemaVersion": 34,
    "style": "dark",
    "tags": [],
    "templating": {
      "list": []
    },
    "time": {
      "from": "now-6h",
      "to": "now"
    },
    "timepicker": {},
    "timezone": "",
    "title": "Mixed graphite default",
    "uid": "B8kpLdZVk",
    "version": 4,
    "weekStart": ""
  }
```
</details>

**Comment by ChristopherzwTan** [comment]
Sorry to bring this topic up again, but just wanted to double-check that if this fix will be ported to Grafana 9.0.x? Or should we just use Grafana 9.1.x that has the fix already.
